### ISSUE TITLE
Potentially change structure of Razer.Devices

### ISSUE THREAD

**Issue by njbmartin** [CoraleStudios/Colore#180]
I've been thinking about this for a while now, but if you have a GUID of a device, it's quite difficult to get a string friendly version of the field name that matches.

This would be particularly useful for #145 

This is literally just a rough thinking, and I know it won't be the right solution, but I've remedied this with:

``` c#
/// <summary>
/// Blackwidow Chroma edition.
/// </summary>
[PublicAPI]
public static readonly Device Blackwidow = new Device
{
    Guid = new Guid(
    0x2ea1bb63,
    0xca28,
    0x428d,
    0x9f,
    0x06,
    0x19,
    0x6b,
    0x88,
    0x33,
    0x0b,
    0xbb),
    Name = "BlackWidow Chroma",
};

/// <summary>
/// Contains device IDs for devices that have Chroma support.
/// </summary>
public class Device
{
    /// <summary>
    /// Gets or sets friendly name of the device.
    /// </summary>
    public string Name { get; set; }

    /// <summary>
    /// Gets or sets Guid for the device.
    /// </summary>
    public Guid Guid { get; set; }
}
```

I also know that this would be a major change, but I just wanted to see what you guys thought about this?

**Comment by Sharparam** [comment]
A way to implement this that would require less drastic changes in the current usage of GUIDs could be to add the names as attributes on the Guid fields like this:

```c#
public static class Devices
{
  [DeviceName("Some Device")]
  public static readonly Guid SomeDevice = new Guid(/* ... */);
}
```

Thoughts?

**Comment by njbmartin** [comment]
So, here's how you could get the DeviceName from the attribute, unless I'm missing something...

**Set the name**
```C#
        [PublicAPI]
        [DeviceName("BlackWidow")]
        public static readonly Guid Blackwidow = new Guid(
            0x2ea1bb63,
            0xca28,
            0x428d,
            0x9f,
            0x06,
            0x19,
            0x6b,
            0x88,
            0x33,
            0x0b,
            0xbb);
```

**Create the attribute**
```C#
    internal class DeviceNameAttribute : Attribute
    {
        private string DeviceName;

        public DeviceNameAttribute(string name)
        {
            this.DeviceName = name;
        }

        public string Name
        {
            get
            {
                return DeviceName;
            }
        }
    }
```

**Retrieve the name from the Guid**
```C#
        public static string GetName(Guid id)
        {
            var fields = typeof(Devices).GetFields();
            foreach (var field in fields)
            {
                if ((Guid)field.GetValue(null) == id)
                {
                    var attr = (DeviceNameAttribute)field.GetCustomAttributes(typeof(DeviceNameAttribute), false)[0];
                    return attr.Name;
                }
            }

            return null;
        }
```

**Comment by njbmartin** [comment]
Custom attribute is useful if we want to give the full name i.e "BlackWidow Tournament Edition", but if we literally just want to find out the field name, we can do this without using a new attribute:

```C#
        public static string GetName(Guid id)
        {
            var fields = typeof(Devices).GetFields();
            foreach (var field in fields)
            {
                if ((Guid)field.GetValue(null) == id)
                {
                    return field.Name;
                }
            }

            return null;
        }
```

**Comment by Sharparam** [comment]
Moving this to v6.1. I'm thinking adding two methods: `GetName(Guid)` and `GetDisplayName(Guid)` to the devices class. What are your thoughts @njbmartin?

**Comment by Sharparam** [comment]
This was implemented early, in bd611194e898b3a0e942cc076f92cb945a9d9ea6.
